VulkanHppGenerator.cpp

```diff
@@ -62,6 +62,53 @@ const std::string exceptionHeader(
   "\n"
   );
 
+std::string const exceptionClassesHeader = (
+  "#if defined(_MSC_VER) && (_MSC_VER == 1800)\n"
+  "# define noexcept _NOEXCEPT\n"
+  "#endif\n"
+  "\n"
+  "  class Error\n"
+  "  {\n"
+  "    public:\n"
+  "    virtual const char* what() const noexcept = 0;\n"
```

Comment:
I added this, because - since we can't inherit from both `std::exception` and `std::system_error` - when user would want to catch all Vulkan exceptions at once, he couldn't do anything with it. Imagine:
```
try {
    // ...
} catch (/* something specific */) {
    // ...
} catch (const vk::Error& vk_error) {
    // What to do with `vk_error`? Can't even print what was the problem...
    // not much more useful then `catch (...)`
}
```
So I think that it would be nice to have at least `what()` available there. If you disagree with this argument I will remove this.

As for virtual destructor - nice catch, I should have seen this... :+1: